packages/legend-shared/src/network/NetworkUtils.ts

```diff
@@ -493,6 +496,21 @@ export const downloadFile = (
   }
 };
 
+export const readFileAsText = (file: File): Promise<string> =>
+  new Promise((resolve, reject) => {
+    const fileReader = new FileReader();
+    fileReader.onload = (): void => {
+      const result = fileReader.result;
+      if (isString(result)) {
+        resolve(result);
+      } else {
+        reject(new UnsupportedOperationError('Unable to read file type'));
```

Comment:
I think since it's Javascript here, feel free to throw `Error` instead, also the message should be `Can't read file`